src/shared/components/tc-communities/JoinCommunity/ConfirmModal/index.jsx

```diff
@@ -35,30 +40,36 @@ export default function ConfirmModal({
           </div>
         )}
       </div>
-      <div styleName="buttons">
+      <div className={style.buttons}>
         { userId ? (
-          <span>
-            <button
+          <div>
+            <PrimaryButton
               onClick={() => join(token, groupIds[0], userId)}
-              styleName="btnConfirm"
-            >Join</button>
-            <button
+              size="sm"
+              theme={theme.primaryBtn}
+              composeContextTheme={COMPOSE.SWAP}
```

Comment:
No need to swap the theme here, and actually no need to pass custom theme into these internal modal buttons the way you do. The themes for these internal buttons should come from the community context; for that you should add `<ThemeProvider>` for each community, like here https://github.com/topcoder-platform/community-app/blob/develop/src/shared/routes/Communities/Veterans/Routes.jsx#L26
So if in that router of Veterans community you add
```
import primaryButtonTheme from 'components/buttons/outline/round/open-sans/blue-uppercase.scss';
```
and then, inside `<ThemeProvider>`'s `theme` object you add the entry
```
PrimaryButton: primaryButtonTheme,
```
Then all instances of `<PrimaryButton>` inside that `<ThemeProvider>` subtree will be styled with this context theme automatically. So, basically, we just need to add such `<ThemeProvider>` to each community, to specify default themes of primary and secondary buttons within each community (which are outline buttons for most of communities you can join yourself, and the default Topcoder buttons for the Veterans).